recipe/meta.yaml

```
@@ -26,17 +32,86 @@ requirements:
     - make  # [unix]
   host:
 
+outputs:
+  # zlib-ng gets built when compat is NOT enabled; versioned with the regular zlib-ng release
+  - name: zlib-ng
+    build:
+      skip: true  # [compat == "yes"]
+  # zlib and libzlib get built when compat is enabled; they are versioned after the zlib API
+  - name: libzlib
+    version: {{ zlib_api }}
+    build:
+      skip: true  # [compat == "no"]
+      string: h{{ PKG_HASH }}_ng_{{ version.replace(".", "") }}_{{ number }}
+      run_exports:
+        # This is for compat mode, just libzlib, with major pinning
+        - {{ pin_subpackage('libzlib', max_pin='x') }}
+      track_features:
+        - zlib-ng-compat
+    files:
+      include:
+        - lib/libz.so.*          # [linux]
+        - lib/libz.*.dylib       # [osx]
+        - Library/bin/zlib.dll   # [win]
+        - Library/bin/zlib1.dll  # [win]
+        - zlib.dll               # [win]
+    test:
+      commands:
+        - test ! -e $PREFIX/include/zlib.h                          # [unix]
+        - test ! -e $PREFIX/lib/libz.a                              # [unix]
+        - test ! -e $PREFIX/lib/libz${SHLIB_EXT}                    # [unix]
+        - if not exist %LIBRARY_BIN%\zlib.dll exit 1                # [win]
+        - if not exist %PREFIX%\zlib.dll exit 1                     # [win]
+        - if not exist %LIBRARY_BIN%\zlib1.dll exit 1               # [win]
+  - name: zlib
+    version: {{ zlib_api }}
+    build:
+      skip: true  # [compat == "no"]
+      number: {{ number }}
+      string: h{{ PKG_HASH }}_ng_{{ version.replace(".", "") }}_{{ number }}
+      run_exports:
+        - {{ pin_subpackage('zlib', max_pin='x') }}
```

This is different than in the zlib recipe.
```suggestion
        - {{ pin_subpackage('libzlib', max_pin='x') }}
```

they should be harmonized IMO.